File: src/main/java/com/tms/csp/transforms/Transforms.java
package com.tms.csp.transforms;

import com.tms.csp.ast.Exp;
import com.tms.csp.ast.PLConstants;

public class Transforms implements PLConstants {


    public static Transformer bnf() {
        return Transformer.BNF;
    }

    public static Transformer pushNotsIn() {
        return Transformer.PUSH_NOTS_IN;
    }

    public static Transformer nnf() {
        return Transformer.NNF;
    }

    public static Transformer identity() {
        return Transformer.IDENTITY;
    }


    public static Exp xorToCnf(Exp in) {
        return new XorsToCnf().transform(in);
    }

    public static Exp removeConflicts(Exp in) {
        return new ConflictsToCnf().transform(in);
    }

    public static Exp removeIffs(Exp in) {
        return new IffToCnf().transform(in);
    }

    public static Exp flattenImps(Exp in) {
        return new FlattenImps().transform(in);
    }

    public static Exp removeImps(Exp in) {
        return new ImpsToCnf().transform(in);
    }

    public static Exp flattenOrs(Exp in) {
        return new FlattenOrs().transform(in);
    }

    public static Exp flattenAnd(Exp in) {
        return new FlattenAnds().transform(in);
    }

    public static Exp unFlattenAnd(Exp in) {
        return andToBinary(in);
    }

    public static Exp andToBinary(Exp in) {
        return new AndsToBinary().transform(in);
    }

    public static Exp pushNotsIn(Exp in) {
        return new PushNotsIn().transform(in);
    }

    public static Exp pushUpAnds(Exp in) {
        return new PushAndsOut().transform(in);
    }


}

File: src/main/java/com/smartsoft/csp/transforms/Transforms.kt
package com.smartsoft.csp.transforms

import com.smartsoft.csp.ast.Exp
import com.smartsoft.csp.ast.PLConstants

open class Transforms : PLConstants {
    companion object {


        fun bnf(): Transformer {
            return Transformer.BNF
        }

        fun pushNotsIn(): Transformer {
            return Transformer.PUSH_NOTS_IN
        }

        fun nnf(): Transformer {
            return Transformer.NNF
        }

        fun identity(): Transformer {
            return Transformer.IDENTITY
        }


        fun xorToCnf(inExp: Exp): Exp {
            return XorsToCnf().transform(inExp)
        }

        fun removeConflicts(inExp: Exp): Exp {
            return ConflictsToCnf().transform(inExp)
        }

        fun removeIffs(inExp: Exp): Exp {
            return IffToCnf().transform(inExp)
        }

        fun flattenImps(inExp: Exp): Exp {
            return FlattenImps().transform(inExp)
        }

        fun removeImps(inExp: Exp): Exp {
            return ImpsToCnf().transform(inExp)
        }

        fun flattenOrs(inExp: Exp): Exp {
            return FlattenOrs().transform(inExp)
        }

        fun flattenAnd(inExp: Exp): Exp {
            return FlattenAnds().transform(inExp)
        }

        fun unFlattenAnd(inExp: Exp): Exp {
            return andToBinary(inExp)
        }

        fun andToBinary(inExp: Exp): Exp {
            return AndsToBinary().transform(inExp)
        }

        fun pushNotsIn(inExp: Exp): Exp {
            return PushNotsIn().transform(inExp)
        }

        fun pushUpAnds(inExp: Exp): Exp {
            return PushAndsOut().transform(inExp)
        }
    }


}
